### PR TITLE
feat: consolidate Insert request related partitioning and distributed processing operations into Inserter

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

same for Delete requests

## What's changed and what's your intention?

* Remove `frontend::instance::distributed::{inserter|deleter}`
* Add `Partitioner` for all `{Insert|Delete}` request converters to partition requests into *region* requests
* Add `handle_table_insert` and `handle_statement_insert` to `frontend::insert::Inserter`
* Add `handle_table_insert` to `frontend::delete::Deleter`
* Remove `handle` in `RegionServerHandler`
* `frontend::insert::Inserter` and `frontend::delete::Deleter` internally send distributed requests

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/frontend/src/insert.rs [hunk 1]
```diff
@@ -78,41 +94,139 @@ impl<'a> Inserter<'a> {
...
+        &self,
+        request: TableInsertRequest,
+        ctx: QueryContextRef,
+    ) -> Result<usize> {
+        let catalog = request.catalog_name.as_str();
+        let schema = request.schema_name.as_str();
+        let table_name = request.table_name.as_str();
+        let table = self.get_table(catalog, schema, table_name).await?;
+        let table = table.with_context(|| TableNotFoundSnafu {
+            table_name: format!("{}.{}.{}", catalog, schema, table_name),
```
Comment (killme2008) [inline src/frontend/src/insert.rs:123]: The same suggestion as above.

File: src/frontend/src/req_convert/delete/column_to_row.rs [hunk 2]
```diff
@@ -35,6 +35,6 @@ fn request_column_to_row(request: DeleteRequest) -> Result<RowDeleteRequest> {
     Ok(RowDeleteRequest {
         table_name: request.table_name,
         rows: Some(rows),
-        region_number: request.region_number,
+        region_number: 0, // deprecated field
```
Comment (killme2008) [inline src/frontend/src/req_convert/delete/column_to_row.rs:38]: Adds a `FIXME` label.

File: src/frontend/src/req_convert/insert/column_to_row.rs [hunk 3]
```diff
@@ -35,6 +35,6 @@ fn request_column_to_row(request: InsertRequest) -> Result<RowInsertRequest> {
     Ok(RowInsertRequest {
         table_name: request.table_name,
         rows: Some(rows),
-        region_number: request.region_number,
+        region_number: 0, // deprecated field
```
Comment (killme2008) [inline src/frontend/src/req_convert/insert/column_to_row.rs:38]: Adds a `FIXME` label.

Comment (killme2008) [reply src/frontend/src/req_convert/insert/column_to_row.rs:38]: Or maybe we can remove it immediately?

**Review by killme2008 — COMMENTED** [review]
Great job!

File: src/frontend/src/req_convert/insert/column_to_row.rs (hunk 3 above)
Comment (fengjiachun) [reply src/frontend/src/req_convert/insert/column_to_row.rs:38]: https://github.com/GreptimeTeam/greptime-proto/pull/101

File: src/frontend/src/insert.rs [hunk 4]
```diff
@@ -78,41 +94,139 @@ impl<'a> Inserter<'a> {
...
+                })
+            });
+        let results = future::try_join_all(tasks).await.context(JoinTaskSnafu)?;
+
+        let affected_rows = results.into_iter().sum::<Result<u64>>()?;
+        counter!(crate::metrics::DIST_INGEST_ROW_COUNT, affected_rows);
+        Ok(affected_rows)
+    }
+
+    async fn group_requests_by_peer(
```
[the remark this answers is not in the document]
Comment (zhongzc) [reply src/frontend/src/insert.rs:179]: I have tried my best to simplify these two functions, but refactoring for the purpose of reusability is a bit complicated due to the limitations of the types. Let's not do it for now.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**